client/my-sites/domains/domain-management/name-servers/dns-templates.jsx

```diff
@@ -38,7 +38,7 @@ class DnsTemplates extends React.Component {
 						}
 					),
 					placeholder: 'google-site-verification=...',
-					validationPattern: /^google-site-verification=\w{43}$/,
+					validationPattern: /^google-site-verification=[A-Za-z0-9_-]{43}$/,
```

Comment:
Well - it works but you can write that regexp as `/^google-site-verification=[\w_-]{43}$/` - it's simpler. Also I would escape the underscore and the dash just for clarity. So `/^google-site-verification=[\w\_\-]{43}$/` :)

Reply:
`\w` is [not a 1-1 substitution for the explicit character class I used](https://www.regular-expressions.info/shorthand.html):
> \w stands for "word character". It always matches the ASCII characters [A-Za-z0-9_]. Notice the inclusion of the underscore and digits. **In most flavors that support Unicode, \w includes many characters from other scripts. There is a lot of inconsistency about which characters are actually included.**

Escaping those characters is not needed either - IMHO, it makes it less readable especially since we have the `\w` one before them (which also include backslash).
In a character classes, only `[`, `^` and `-` have a special meaning and the last one only when between characters. I prefer to avoid escaping characters there when possible, because it makes the regular expression (which is already a bit cryptic) even harder to read.
`[A-Za-z0-9_-]` vs `[A-Za-z0-9\_\-]` - the unnecessary escaping makes it not clear to me why the backslash got there - maybe it was mean to be a literal backslash (`\\`)? Just to be clear - these rules are consistent between different flavours of regexps, so once you learn them, you can apply them always.

Reply:
Huh - TIL that ` \w` includes the underscore. I personally prefer the shorter version and escaping any literal character (especially `-`) inside a character class definition because it has a specific meaning.

Anyway `\-` if that works for you I don't mind :D

Reply:
I see what you did there :trollface: 

> TIL that \w includes the underscore

The way I understand the documentation, it _can_ also include stuff like Unicode equivalents of letters (`example` vs `ｅｘａｍｐｌｅ` or `𝚎𝚡𝚊𝚖𝚙𝚕𝚎`). But on Google's site, I'm guessing they only allow ASCII.